### ISSUE TITLE
[enhancement]: As a user I want to open Gitpod on a particular Git commit

### ISSUE THREAD

**Issue by kittaakos** [gitpod-io/gitpod#793]
I wanted to open gipod.io on this commit:
 - https://github.com/eclipse-theia/theia/commit/9d9b3725cb20b5a50d2aeadf69fd84e46ae016d5.
 - I prefixed the URL with `gitpod.io/#`.
 - I have opened the workspace. The UI said the workspace is not running. The error was most likely a temporary issue, so I navigated to my dashboard.
 - The dashboard reported two different commit hashes for my workspace.

![Screen Shot 2019-09-16 at 08 47 44](https://user-images.githubusercontent.com/1405703/64939412-2ecb1800-d861-11e9-824c-5bb771763e06.jpg)

Long story short, I would like to open Gitpod on particular commit not just on a branch or a PR.
